src/cata_tiles.cpp

```diff
@@ -972,51 +962,156 @@ void cata_tiles::draw_rhombus(int destx, int desty, int size, SDL_Color color, i
     }
 }
 
+//converts a submap x/y location to a string with the contents "x_y"
+//should be unique enough to track submaps ingame
+static std::string get_submap_id(const tripoint &t)
```

Comment:
You shouldn't need this, you can store a point as key (instead of the string) in the map. You could perhaps even use the tripoint directly. But if not, the conversion from tripoint to point is trivial and can be done directly without a separate function.

Also, consider using one of the conversion functions of the overmap buffer, e.g. `overmapbuffer::ms_to_sm_copy`, which converts from map squares (ms) to submap coordinates (sm). It's about the same as your `x / SEEX`, but it makes the intent clear (and it avoids a bug: `-1/SEEX` is 0, as is `+1/SEEX`, but the map coordinates (-1,0) and (+1,0) are actually on different submaps).